README.md

```diff
@@ -57,15 +57,55 @@ The usage and options of `node-watch` are compatible with [fs.watch](https://nod
 
     // filter with custom function
     watch('./', { filter: f => !/node_modules/.test(f) });
+
+    ```
+
+   Each file and directory will be passed to the filter to determine whether
+   it will then be passed to the callback function. Like `Array.filter` does in `JavaScript`.
+   There are three kinds of return values for filter function:
+
+     * **`true`**: Will be passed to callback.
+     * **`false`**: Will not be passed to callback.
+     * **`skip`**: Same with `false`, and skip to watch all its subdirectories.
+
+
+   On Linux, where the `recursive` option is not natively supported,
+   you could skip to watch all its subdirectories by returning the `skip` flag.
+
+    ```js
+    watch('./', {
+      resursive: true,
+      filter(f, skip) {
+        // skip node_modules
+        if (/\/node_modules/.test(f)) return skip;
+        // skip .git folder
+        if (/\.git/.test(f)) return skip;
+        // only watch for js files
+        return /\.js$/.test(f);
+      }
+    });
+
     ```
+
+    Use [minimatch](https://www.npmjs.com/package/minimatch) for filter:
```

Comment:
Might be worth promoting [picomatch](https://www.npmjs.com/package/picomatch) here as fairly popular dependency-free alternative to micromatch ([example in QUnit](https://github.com/qunitjs/qunit/commit/51fe12c4ef04931497)).

Compared:

```js
// re-used
const { Minimatch } = minimatch;
mm = new Minimatch(…);
mm.match(filepath); // bool

// once
minimatch(filepath, pattern); // bool
```


```js
// re-used
isMatch = picomatch(…);
isMatch(filepath); // bool

// once
picomatch.isMatch(filepath, pattern); // bool
```

Reply:
Updated! Hope they won't be hard to understand despite of my poor English.